backend/api/tasks/statistics.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+from flask_restful import Resource, current_app, request
+from backend.services.stats_service import NotFound, StatsService
+
+
+def validate_date_input(date):
+    if date:
+        date_format = "%Y-%m-%d"
+        current_date = datetime.datetime.now()
+        date = datetime.datetime.strptime(date, date_format)
+        if date <= current_date:
+            return date
+        raise Exception("Error: Date out of range for task activity")
+
+
+class TasksStatisticsAPI(Resource):
+    def get(self):
+        """
+        Get Task Stats
+        ---
+        tags:
+          - tasks
+        produces:
+          - application/json
+        parameters:
+            - in: query
+              name: start_date
+              description: Date to filter as minimum
+              required: false
+              type: string
+              default: null
+            - in: query
+              name: end_date
+              description: Date to filter as maximum
+              required: false
+              type: string
+              default: null
+            - in: query
+              name: organisationName
+              description: Organisation name to filter by
+              required: false
+              default: null
+            - in: query
+              name: organisationID
+              description: Organisation ID to filter by
+              required: false
+              default: null
+            - in: query
+              name: campaigns
+              description: Campaign name to filter by
+              required: false
+              default: null
+            - in: query
+              name: project_id
+              description: Project IDs to filter by
+              required: false
+              default: null
+            - in: query
+              name: country
+              description: Country name to filter by
+              required: false
+              default: null
+        responses:
+            200:
+                description: Task statistics
+            404:
+                description: Not found
+            500:
+                description: Internal Server Error
+        """
+        try:
+            start_date = validate_date_input(request.args.get("start_date"))
+            end_date = validate_date_input(request.args.get("end_date"))
+            org_id = request.args.get("org_id", None, int)
+            org_name = request.args.get("org_name", None, str)
+            campaign = request.args.get("campaign", None, int)
+            project_ids = request.args.get("project_ids")
```

Comment:
Please use the write query parameter for project ID. `project_ids` -> `project_id`

Reply:
All the argument names should match the YAML query parameter names, without which values would be `None`